assessments/projects/contour/self-assessment.md

```diff
@@ -0,0 +1,206 @@
+When a request is sent, first it is routed by a load-balancer to an instance of Envoy. The Envoy proxy then sends the request to a Contour pod.
+
+
+### Goals
+Contour serves as a Layer 7 HTTP middleware reverse proxy for enabling ingress to Kubernetes clusters.
+
+### Non-goals
+Contour is not a service mesh, nor is the intention for Contour to expose all of the features or configuration options of Envoy. 
+Contour is not intended for proxying layer 4 protocols such as TCP or UDP except when needed to deliver HTTP.
+TCP proxying is available but expects the proxying to be passed through HTTPS. 
+Using Contour to proxy raw TCP or UDP traffic may work, but is not the intended usage.
+
+## Self-assessment use
+
+This self-assessment is created by Heejin Jang, Justin Rivera, Samuel Vieira Restrepo, and Esther Wang 
+to perform an internal analysis of the project's security. It is not intended to provide 
+a security audit of Contour, or function as an independent assessment or attestation of Contour's security health.
+
+This document serves to provide Contour users with an initial understanding of
+Contour's security, where to find existing security documentation, Contour plans for
+security, and general overview of Contour security practices, both for development of
+Contour as well as security of Contour.
+
+This document provides the CNCF TAG-Security with an initial understanding of Contour
+to assist in a joint-assessment, necessary for projects under incubation.  Taken
+together, this document and the joint-assessment serve as a cornerstone for if and when
+Contour seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical
+  * xDS server – Functions as a control plane for Envoy. The xDS server knows all the secrets that Contour knows, so Contour requires a mutual TLS authentication between Contour and Envoy such that another cluster cannot simply connect to the xDS server and collect all the known secrets.
+* Security Relevant
+  * Federal Information Processing Standard (FIPS) 140-2 – FIPS describes United States government approved secuirty requirements for cryptographib modules. Software validated by an accredited Crpytographic Module Validation Program (CVMP) labortory is suitable for use in applications for US governmental departments or industries subjected to US Federnal regulations. Contour is not FIPS validated but can be built and configured to the standards FIPS 140-2 establishes.
+
+## Project compliance
+
+* Compliance
+  * TBD after discussion with project maintainers
+
+## Secure development practices
+
+* Development Pipeline - Contour utilizes GitHub issues, labels, and code reviews for secure open source development.
```

Comment:
Can you elaborate what the development pipeline looks like? is it automated, triggered for every change introduced to repo, what kind of security tools integrated into this pipeline etc? The reference shared here doesn't seem to capture them

Reply:
After investigating a bit, we've added some automated checks/rules + guidelines used for secure development & included more references